src/Symfony/Component/Cache/Marshaller/DefaultMarshaller.php

```diff
@@ -81,6 +81,14 @@ public function unmarshall(string $value): mixed
             }
 
             throw new \DomainException(error_get_last() ? error_get_last()['message'] : 'Failed to unserialize values.');
+        } catch (\UnserializationFailedException $e) {
+            $p = $e->getPrevious();
+
+            if ($p instanceof \Error) {
+                throw new \ErrorException($p->getMessage(), $p->getCode(), \E_ERROR, $p->getFile(), $p->getLine());
+            }
+
+            throw $p instanceof \DomainException ? $p : $e;
```

Comment:
The method throws a `DomainException` when a class referenced by the serialized payload is not found, and `Error` are wrapped into `ErrorException`. Unwrapping `UnserializationFailedException` is required to restore both behaviors.